### PR TITLE
Check parallel

### PR DESCRIPTION
**Pull request by Ovascos**
make check is executing tests in parallel (when either GNU parallel or moreutils' parallel is available)

### CODE REVIEW

**Comment by Ovascos** [comment]
Note that this PR is tested with Mac and Linux only. I don't have a Windows+cygwin setup available.

**Comment by ahmed-irfan** [comment]
How do you specify the number of cores to use?

**Comment by Ovascos** [comment]
You can not. All cores are used.

File: tests/regress/run_test.sh
Comment (ahmed-irfan) [inline tests/regress/run_test.sh:1]: is this code essentially coming from the check.sh? If yes, are there any changes?

**Comment by ahmed-irfan** [comment]
> You can not. All cores are used.

Is there an easy way to pass this number?

**Comment by Ovascos** [comment]
> > You can not. All cores are used.
> 
> Is there an easy way to pass this number?

I'll have a look.

**Comment by Ovascos** [comment]
Integrated check.sh within GNU make's token system to use limit the number of total parallel execution to the value specified by -j N

**Review by ahmed-irfan — APPROVED** [review]
LGTM

Thanks

**Pull request merged**